python/ray/_private/gcs_utils.py

```diff
@@ -302,6 +319,114 @@ def pin_runtime_env_uri(self, uri: str, expiration_s: int) -> None:
             )
 
 
+class GcsAioClient:
+    def __init__(
+        self,
+        channel: Optional[GcsChannel] = None,
+        address: Optional[str] = None,
+    ):
+        if channel is None:
+            assert isinstance(address, str)
+            channel = GcsChannel(gcs_address=address, aio=True)
+        assert isinstance(channel, GcsChannel)
+        assert channel._aio is True
+        self._channel = channel
+        self._connect()
+
+    def _connect(self):
+        self._channel.connect()
+        self._kv_stub = gcs_service_pb2_grpc.InternalKVGcsServiceStub(
+            self._channel.channel()
+        )
+
+    async def internal_kv_get(
+        self, key: bytes, namespace: Optional[bytes], timeout: Optional[float] = None
+    ) -> Awaitable[bytes]:
+        logger.debug(f"internal_kv_get {key} {namespace}")
+        req = gcs_service_pb2.InternalKVGetRequest(namespace=namespace, key=key)
+        reply = await self._kv_stub.InternalKVGet(req, timeout=timeout)
```

Comment:
I just noticed this "await" - doesn't this mean this statement blocks until the result is ready? So why should the method be async (line 342) in the first place?